@here/harp-transfer-manager/src/TransferManager.ts

```diff
@@ -85,21 +85,25 @@ export class TransferManager implements ITransferManager {
         init?: RequestInit
     ): Promise<Response> {
         try {
-            const response = await fetchFunction(url, init);
-            if (response.status !== 503 || retryCount > maxRetries) {
```

Comment:
The `retryCount > maxRetries` is also wrong, because `retryCount` is 0 based, and we wait to get past `maxRetries`, so we actually get 2 more than maxRetries, i.e. if maxRetries is 3, then retryCount is 0,1,2,3,4, i.e. 5 attempts, hence this is also fixed.